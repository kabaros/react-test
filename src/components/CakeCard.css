.cake-card {
    width: 100%;
    border: 1px solid #f3f3f3;
    margin: 1rem;
}

.cake-card .card-header {
    width: 100%;
}

.cake-card img {
  width: 100%;
  height: 200px;
}

@media (min-width: 760px) {     
    .cake-card {
        width: 40%;
    }
}


@media (min-width: 1000px) {     
    .cake-card {
        width: 30%;
    }
}