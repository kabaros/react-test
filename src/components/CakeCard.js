import React from 'react';
import PropTypes from 'prop-types';
import './CakeCard.css';

const CakeCard = ({ cake }) => {
    return (<article className="cake-card">
        <div className="card-header">
            <img alt={cake.title} src={cake.image} />
        </div>
        <h2>{cake.title}</h2>
        <p>{cake.desc}</p>
    </article>);
};

CakeCard.propTypes = {
    cake: PropTypes.shape({
        title: PropTypes.string,
        desc: PropTypes.string,
        image: PropTypes.string
    })
};

export default CakeCard;