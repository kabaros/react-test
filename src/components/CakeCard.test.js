import React from 'react';
import ReactDOM from 'react-dom';
import CakeCard from './CakeCard';

it('renders a card with a cake', () => {
  const div = document.createElement('div');
  const cake = {
      desc: 'some description',
      img: 'some-image-url',
      title: 'image-title'
  };
  ReactDOM.render(<CakeCard cake={cake} />, div);
  expect(div).toMatchSnapshot();
});
