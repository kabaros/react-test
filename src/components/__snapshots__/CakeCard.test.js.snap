// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Search box for matching cakes 1`] = `
<article
  className="cake-card"
>
  <div
    className="card-header"
  >
    <img
      alt="image-title"
      src={undefined}
    />
  </div>
  <h2>
    image-title
  </h2>
  <p>
    some description
  </p>
</article>
`;
